src/util/strencodings.h

```diff
@@ -67,6 +68,66 @@ std::vector<Byte> ParseHex(std::string_view hex_str)
 {
     return TryParseHex<Byte>(hex_str).value_or(std::vector<Byte>{});
 }
+
+/** consteval version of HexDigit() without the lookup table. */
+consteval uint8_t ConstevalHexDigit(const char c)
+{
+    if (c >= '0' && c <= '9') return c - '0';
+    if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
+
+    throw "Only lowercase hex digits are allowed, for consistency";
+}
+
+struct ConstevalHexLiteral {
+    const std::string_view inner;
+    template <size_t N>
+    consteval ConstevalHexLiteral(const char (&hex_str)[N]) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(const std::string_view hex_str) : inner{hex_str} { Validate(inner); }
+    consteval ConstevalHexLiteral(std::nullptr_t) = delete;
+
+    static consteval void Validate(const std::string_view hex_str)
+    {
+        std::ranges::for_each(hex_str, ConstevalHexDigit);
```

Comment:
super-nit (feel free to ignore): we might want to do the content checking after the size check, it's cheaper

Reply:
Done in latest.